capz/readme.md

```diff
@@ -4,44 +4,96 @@ These scripts and templates are used with [cluster-api-provider-azure](https://g
+
+First clone and configure the following repositories
+
+### CAPZ repo setup
+
+Clone the [cluster-api-provider-azure](https://github.com/kubernetes-sigs/cluster-api-provider-azure) and checkout / sync out the latest release branch.
+
+### Cloud Provider Azure repo setup
+
+Clone the [cloud-provider-azure](https://github.com/kubernetes-sigs/cloud-provider-azure) and checkout / sync to the correct branch
+
+- if testing kubernetes@master then checkout / sync to the master branch
+- if testing a released version of Kubernetes then checkout / sync the corresponding release branch in the cloud-provider-azure repo
+
+> Note: To run e2e tests with the same configurations as the upstream e2e test passes, look at the `extra_refs` section of the **ci-kubernetes-e2e-capz-master-windows** in [release-master-windows.yaml](https://github.com/kubernetes/test-infra/blob/master/config/jobs/kubernetes-sigs/sig-windows/release-master-windows.yaml) to see which branches the SIG-Windows e2e test passes are using during the periodic jobs.
+
+### Set environment variables
+
+#### Required
 
 ```bash
 export AZURE_SUBSCRIPTION_ID=<sub-id>
 export AZURE_CLIENT_ID=<client-id>
 export AZURE_CLIENT_SECRET=<client-secret>
 export AZURE_TENANT_ID=<tenantid>
-export CAPZ_DIR="$HOME/<path-to-capz>/cluster-api-provider-azure
+export CAPZ_DIR="$HOME/<path-to-repo>/cluster-api-provider-azure"
+export AZURE_CLOUD_PROVIDER_ROOT="$HOME/<path-to-repo>/cloud-provider-azure"
+
+# run-capz-e2e.sh builds and publishes cloud-provider-azure container images
+# to the registry specified below.  Ensure you are logged in to the registry
+# prior to running the script!
+# TODO: Figure out how to use pre-built images here...
+export REGISTRY="<registry>"
 
 # optional for ability to use your own ssh key (otherwise it generates one)
 # NOTE: Azure does not support ed25519 encrypted SSH keys!
 export AZURE_SSH_PUBLIC_KEY_FILE="$HOME/.ssh/id_rsa.pub"
 ```
 
-## Other configuration
+#### Optional
 
 | ENV variable  | Description  |
 | ------------- | ------------ |
-| `SKIP_CREATE` | Don't create a cluster.  Must set `CLUSTER_NAME` and have current a workload cluster kubeconfig file with name `./"${CLUSTER_NAME}".kubeconfig` |
-| `SKIP_LOG_COLLECTION` | Don't collect logs from the cluster |
-| `SKIP_TEST`  | Only creates the cluster, will not run tests |
-| `SKIP_CLEANUP` | Don't delete the cluster / resource group after script executions |
+| `API_SERVER_FEATURE_GATES` | Comma-separated list of feature-gates and their values to pass to the kube-apiserver (Defaults to "") |
+| `AZURE_LOCATION` | The azure region to deploy resources into. If not specified a random region will be selected) |
+| `KUBERNETES_VERSION`  | Valid values are `latest` (default) and  `latest-1.xx` where x is valid kubernetes minor version such as `latest-1.24` |
+| `NODE_FEATURE_GATES` | Comma-seperated list of feature-gates and their values to pass to the kubelet (Defaults to "HPAContainerMetrics=true") |
```

Comment:
Does `NODE_FEATURE_GATES` get applied to the kubelet configuration of control plane nodes? I am not seeing it take effect.

Reply:
no, let me fix that up

Reply:
I can confirm that it does not take effect. The template would need to be updated to add the feature gate and cli option to the kubelet.